src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,829 @@
+        # Note: sys log/print will not display in terminal, for debugging comment __remove_tmp_file_reset_stdout and check the tmp file
+        self.__create_tmp_file_for_log_and_set_stdout()
+
+    def tearDown(self):
+        # Reset sys.stdout, close and delete tmp file
+        # Note: sys log/print will not display in terminal, for debugging comment __remove_tmp_file_reset_stdout and check the tmp file
+        self.__remove_tmp_file_reset_stdout()
+
+        self.runtime.stop()
+
+    def test_assessment_patches_under_size_limit_not_truncated(self):
+        """ Perform no truncation on assessment patches.
+        Input: xxx assessment patches
+        Output:
+        operation: Assessment,
+        assessment substatus name: PatchAssessmentSummary,
+        assessment substatus status: success,
+        assessment substatus patches == patch_count,
+        assessment errors code: 0 (success),
+        assessment errors details count: 0,
+        assert count of patches removed from log,
+        completed status file < 126kb,
+        size of status file before truncation == size of status file after truncation. """
+
+        self.runtime.execution_config.operation = Constants.ASSESSMENT
+        self.runtime.status_handler.set_current_operation(Constants.ASSESSMENT)
+
+        patch_count = 500
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count)
+        self.runtime.status_handler.set_package_assessment_status(test_packages, test_package_versions)
+        self.runtime.status_handler.set_assessment_substatus_json(status=Constants.STATUS_SUCCESS)
+        self.runtime.status_handler.log_truncated_patches()
+
+        # Assert complete status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.complete_status_file_path, 'r') as file_handle:
+            complete_substatus_file_data = json.load(file_handle)
+
+        self.__assert_patch_summary_from_status(complete_substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_SUCCESS, patch_count, under_internal_size_limit=True, is_truncated=False)
+
+        # Assert no truncated status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)
+
+        self.assertEqual(len(json.dumps(substatus_file_data).encode('utf-8')), len(json.dumps(complete_substatus_file_data).encode('utf-8')))  # Assert both files have same bytes
+        self.__assert_patch_summary_from_status(substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_SUCCESS, patch_count, under_internal_size_limit=True, is_truncated=False)
+
+        # Assert 'Count of patches removed from: [Assessment=0] [Installation=0] log message is called
+        self.__read_tmp_file_and_assert_log_msg(substatus_file_data, patch_count_assessment=patch_count)
```

Comment:
We can easily get the count of patches removed from self.runtime.status_handler.__assessment_patches_removed and self.runtime.status_handler.__installation_patches_removed.

Please remove the log read logic and fetch patches removed count from status_handler object in this test suite

Reply:
Applies in all other places this is used

Reply:
I don't think it's possible to directly use status_handler to fetch removed count variables because those variables are private, unless we are okay with using reflection but koshy is against this because any name change will cause a test failure, maintenance cost is too high. or if we are okay making those variables public (I don't like this approach, we should encapsulate all variables that are class level usage)

this is why I have to use a public function to do the log and check the log output

Reply:
@feng-j678 How long does this test take to execute? Adding a logic to fetch items from logs doesn't feel like the most efficient way to test simple variable values. Instead we can add functions to simply fetch the variables

Reply:
it doesn't take too long, per ut runs like avg 300ms ish. it will be nice to have public getter method for private truncation variable, it will be easier to set up ut and validate patch count assertions. i'm not sure if this approach we will be able code cover this log output because this is just a log no return type whatsoever. 
self.composite_logger.log_debug("Count of patches removed from: [Assessment={0}] [Installation={1}]".format(len(self.__assessment_patches_removed), len(self.__installation_patches_removed)))

done